command/history/history.go

```diff
@@ -36,6 +38,7 @@ func Bindables(_ command.Commander, _ gxui.Driver, theme *basic.Theme) []bind.Bi
 type History struct {
 	current *node
 	skip    []input.Edit
+	mu      sync.Mutex
```

Comment:
I'm going to take a look through this code to see if we can solve the problem with atomic operations instead of a mutex, just to make sure everything's as fast as possible.

Mutexes are fine in many places, but the History hooks (at least) are triggered every time any text is entered.  Since text is entered frequently, I'm trying to avoid using mutexes and channels whenever possible in any code that happens every time text is entered.

Reply:
Yes, alternative would be much better since with this patch difference in speed significant (i.e. I can see it without benchmarking).